.circleci/config.yml

```diff
@@ -520,21 +518,45 @@ workflows:
     jobs:
       - merge_pr_to_develop
 
-  nightly_build:
+  # Python versions that are supported on `main`.
+  nightly_build_main:
     triggers:
       - schedule:
           cron: 30 2 * * *
           filters:
             branches:
               only: main
     jobs:
-      - build_docker_image
+      - build_docker_image:
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
       - build_kedro:
           matrix:
             parameters:
               python_version: ["3.6", "3.7", "3.8"]
           requires:
-            - build_docker_image
+            - build_docker_image-<<matrix.python_version>>
+
+  # Python version that are *only* supported on `develop`.
+  nightly_build_develop:
```

Comment:
This needs to be done as a separate workflow since it needs to use the requirements on `develop` branch. Just using the ones from `main` doesn't pass the `build_kedro-3.9` job.